aggregators/rules/cr.js

```diff
@@ -1,16 +1,96 @@
+var request = require("request");
+var console = require("console");
+
+const CR_ADDRESS = process.env.CR_ADDRESS; // http://media.wizards.com/2016/docs/MagicCompRules_20160930.txt or newer
+
 class CR{
     constructor(){
-        this.Location = "http://blogs.magicjudges.org/rules/cr/";
+		this.location = "http://blogs.magicjudges.org/rules/cr/";
+		request({url: CR_ADDRESS, encoding: "utf16le"}, (error, response, body) => {
+			 if (!error && response.statusCode == 200) {
+				 let {cr, glossary} = parseCR(body);
+				 this.cr = cr;
+				 this.glossary = glossary;
+				 console.log("CR Ready")
+			 } else {
+				 console.error(error);
+			 }
+		});
     }
     find (parameter,callback){
-        //todo
+        callback(this.cr.get(parameter.trim().replace(/\.$/, "").toLowerCase()));
     }
     getContent(parameter,callback) {
         if (parameter) {
-            this.find(parameter, callback());
+            this.find(parameter, callback);
         } else {
-            callback(this.Location);
+            callback(this.location);
         }
     }
+	getGlossaryEntry(parameter, callback) {
+		callback(this.glossary.get(parameter.trim().toLowerCase()));
+	}
 }
+
+
+function parseCR(crText) {
```

Comment:
These functions aren't visible outside the cr module. Defining them as methods on CR would make them visible outside of the module, which is not really necessary. Defining them elsewhere (inside the constructor or inside a closure around the class definition) would just reduce readability for very little gain.